### DISCUSSION TITLE
Subjects (and hashtags)

### DISCUSSION THREAD

**Discussion by joric3** [bookwyrm-social/bookwyrm#2974] [Ideas]
I was wondering what I'm getting out of the subjects listed for books. I can't search for specific subjects or click on a given subject to see what books have this particular subject as well. It would be nice to get more out the subject fields, either via advanced search options (search subject fields; browse all subjects used on my home instance) or when viewing a book being able to click a subject and get a list of books with this subject. The latter option is similar to hashtags. It would also be cool if I could follow/ subscribe to a subject.

When creating or editing a book, I can enter basically anything. It would be nice if there was some way to see existing subjects so I can better choose which subjects to list for the book I'm dealing with. (This was also mentioned in #2743)

Since I'm reading in German and English mostly, I'm interacting with both German and English subjects, some of which may be more or less equivalent. Even within a single language, some subjects may be interchangeable (SF, SciFi, Sci-Fi, Science Fiction, and probably the same without uppercase, depending on how sensitive to capitalization the system is). I haven't checked whether subjects are saved for a book/work or only for an edition of a book, but users will likely enter subjects in the language of the edition they're reading, even though the subjects of a work are language-independent. It would be nice if subjects could be multiformal, that is: A subject might have alternative forms: SF, scifi, fantascienza, bilimkurgu, ... (which, according to Wikipedia, are the Italian and Turkish terms for science fiction). Instead of juggling these alternatives ourselves, maybe an external source like Wikidata could be used. Instead of entering a freeform string into a subject field, I'd have to enter a Wikidata ID, e.g. Q24925 for science fiction, to stick with this example: https://www.wikidata.org/wiki/Q24925

My thoughts haven't boiled down into a focused feature request, so I'm starting a discussion for now. The second half of my post (language-independent subjects) seems difficult and unlikely, but the first bit about being able to search subjects and maybe and follow them feels less far-fetched. How do others use subjects, and what functions would you like them to serve?

**Comment by teusday** [comment]
I think the issue I just opened for filtering by genre ties into this ([here](https://github.com/bookwyrm-social/bookwyrm/issues/3069)). I like the wikidata cross-referencing for solving issues of internationalization and normalization (so there's not "Science Fiction", "Fiction, Science", etc.)
